src/rtpi.h

```diff
@@ -12,9 +12,19 @@
 
 #include "rtpi_internal.h"
 
+#ifdef __cplusplus
+extern "C" {
```

Comment:
I went with __BEGIN_DECLS/__END_DECLS from 'sys/cdef.h' in https://github.com/gratian/librtpi/commit/ae5d379918e1904f4f515cfab6574afb7d695a4a
Happy to post a PR for the commit above if we like that better.

Reply:
Is `<sys/cdefs.h>` a glibc-ism? It doesn't seem to be part of gcc, and searching around it doesn't seem to be included in musl, which would impact users trying to use librtpi+musl. This may be a theoretical thing-- I don't know how many actual users exist that fit in that category, but also it seems straightforward enough for us to just do the `#ifdef __cplusplus` ourselves here.

Reply:
Yeah I think `<sys/cdefs.h>` might be a glibc thing. I see your point and I agree using the `#ifdef __cplusplus` is more portable. Ignore my previous comment.